homeassistant/components/insteon/__init__.py

```diff
@@ -465,6 +465,11 @@ def group(self):
         """Return the INSTEON group that the entity responds to."""
         return self._insteon_device_state.group
 
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self.name
```

Comment:
This is weird, we should just remove self.name as it's confusing for users and move the definition to unique_id property. Don't we have access to a real name for the name property?

Reply:
From an Insteon stand point **name** gets the underlying device id with group for multi referenced nodes (button id or sub device) and is the "real name".  It is also unique within an Insteon network and can act as the uinique_id for the registry.

Do the **name** and **unique_id** need to be different?

Reply:
Maybe we can get a device description, model, and Insteon address kinda like zwave does for its nodes.

So name would become something like 
"Keypad Dimmer 2334-222 291abb_2",
"I/O Linc 2450 440f9c"
"Door Sensor 2843-232 21545c"

This ends up looking like this
![Example](http://wonderslug.s3-website-us-east-1.amazonaws.com/snaps/Home_Assistant_2018-10-05_18-37-18.png)

Reply:
Of note doing this does make a breaking change for entity_ids being different for existing customizations.

Reply:
We don't need any of those node IDs anymore, once there is unique ID, people can change it. And it will also make sure there is never a collision of IDs, they get reserved.

So name should be the most natural name one can think of. Yes, that will be a breaking change so we should do this in 2 phases: first just introduce the unique ID that is the same as the name which tries to be unique.

Then in a couple of releases, we can drop the name being all weird, because everyone who has upgraded already has their entity ID locked in, making it a non-breaking change.

Reply:
Still I want to make sure we copy the implementation of name property because I don't trust that in the future someone will jsut change the name, messing with the unique IDs, and now that would suck.